server/src/main/java/org/elasticsearch/indices/ShardLimitValidator.java

```diff
@@ -206,16 +206,44 @@ private Result checkShardLimitOnBothGroups(int newShards, int newFrozenShards, C
      * @param state             The cluster state, used to get cluster settings and to get the number of open shards already in the cluster
      */
     public static Result checkShardLimitForNormalNodes(int numberOfNewShards, int replicas, ClusterState state) {
```

Comment:
Since we have 2 new methods to check the shard limit could we remove this one (it's only used in one place) and replace its usage with `checkShardLimitForNormalNodes` ?